cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -120,13 +121,13 @@ protected ProgramType getProgramType(String programType) {
     }
   }
 
-  protected ProgramRuntimeService.RuntimeInfo findRuntimeInfo(Id.Program programId,
+  protected ProgramRuntimeService.RuntimeInfo findRuntimeInfo(ProgramId programId,
                                                               ProgramRuntimeService runtimeService) {
     Collection<ProgramRuntimeService.RuntimeInfo> runtimeInfos = runtimeService.list(programId.getType()).values();
     Preconditions.checkNotNull(runtimeInfos, UserMessages.getMessage(UserErrors.RUNTIME_INFO_NOT_FOUND), programId);
 
     for (ProgramRuntimeService.RuntimeInfo info : runtimeInfos) {
-      if (programId.equals(info.getProgramId())) {
+      if (programId.equals(info.getProgramId().toEntityId())) {
```

Comment:
let's do that in a separate PR. may have to cherry-pick it to 3.6 too if required.